File: Samples/C++/OpenSaveCloseProject/OpenSaveCloseProject.cpp
#include <iostream>
#include <conio.h>
#include <string>

#import <RengaCOMAPI.tlb>

#include <Renga/CreateApplication.hpp>


int main(int argc, char** argv)
{
  if (argc != 2)
  {
    std::cout << "usage: " << argv[0] << " ProjectFullFilePath\n";
    return 0;
  }

  ::CoInitialize(nullptr);

  // Creating Renga
  auto renga = Renga::CreateApplication(CLSCTX_LOCAL_SERVER);
  renga->PutVisible(VARIANT_TRUE);
  int result = renga->OpenProject(bstr_t(argv[1]));
  if (result)
  {
    std::cout << "Open file '" << argv[0] << "' error\n";
    renga->Quit();
    return 0;
  }

  auto pProject = renga->GetProject();

  // Saving project as
  pProject->Save();

  wchar_t buffer[MAX_PATH];
  ::GetCurrentDirectory(MAX_PATH, buffer);

  const auto filePathToSave = std::wstring(buffer) + L"/project.rnp";
  
  // Saving project as
  pProject->SaveAs(filePathToSave.c_str(), Renga::ProjectType::ProjectType_Project, VARIANT_FALSE);

  // Closing project
  renga->CloseProject(VARIANT_TRUE);

  // Closing Renga
  renga->Quit();

  ::CoUninitialize();
}

File: Samples/Python/OpenSaveCloseProject.py
import sys
import os
import win32com.client


if __name__ == '__main__':

    if len (sys.argv) == 1:
        print("usage: {} sourceProjectFileName".format(os.path.split(__file__)[1]))
        sys.exit(1)

    app = win32com.client.Dispatch("Renga.Application.1")
    app.Visible = True

    projectFilePath = sys.argv[1]
    print("Opening project: " + projectFilePath)

    result = app.OpenProject(projectFilePath)
    if result != 0:
        print("Error opening project")

    print("Getting project")
    project = app.Project
    if project == None:
        print("Error: project equals None")

    if len (sys.argv) == 2:
        print("Saving project")
        result = project.Save();
        if result != 0:
            print("Error saving project")
    else:
        saveAsFilePath = sys.argv[2]
        print("Saving project as {}".format(saveAsFilePath))
        result = project.SaveAs(saveAsFilePath, 0, 1);
        if result != 0:
            print("Error saving project as")

    print("Closing project")
    result = app.CloseProject(1)
    if result != 0:
        print("Error closing project")

    print("Quiting application")
    app.Quit()
